### PR TITLE
Add java and composite steps autocompletion

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/finder/StepDefinitionFinder.java [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+import org.vividus.studio.plugin.util.RuntimeWrapper;
+
+@Singleton
+@SuppressWarnings("restriction")
+public class StepDefinitionFinder implements IStepDefinitionFinder
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(StepDefinitionFinder.class);
+    private static final Pattern PARAMETER_NAME_PATTERN = Pattern.compile("\\$\\w+");
+    private static final Pattern STEP_ANNOTATION_PATTERN = Pattern
+            .compile("^org\\.jbehave\\.core\\.annotations\\.(When|Then|Given)$");
```
Comment (valfirst) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/finder/StepDefinitionFinder.java:69]: `@Alias` as well

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/model/StepType.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    public String getId()
+    {
+        return this.id;
+    }
+
+    public static StepType detect(String stepAsString)
+    {
+        String prefix = StringUtils.substringBefore(stepAsString, " ");
+        return Arrays.stream(StepType.values())
```
Comment (valfirst) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/model/StepType.java:52]: I recommend to apply checkstyle from vividus-build-system, where `Stream.of` is forced to be used

File: vividus-studio-server/vividus-studio-plugin/pom.xml [hunk 3]
```diff
@@ -14,8 +14,35 @@
     <properties>
         <maven.compiler.source>12</maven.compiler.source>
         <maven.compiler.target>12</maven.compiler.target>
+        <junit.jupiter.version>5.6.2</junit.jupiter.version>
+        <org.mockito.version>3.3.3</org.mockito.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <checkstyle.version>8.33</checkstyle.version>
```
Comment (valfirst) [inline vividus-studio-server/vividus-studio-plugin/pom.xml:20]: fyi: https://github.com/checkstyle/checkstyle/issues/8299

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/finder/StepDefinitionFinder.java (hunk 1 above)
Comment (uarlouski) [reply vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/finder/StepDefinitionFinder.java:69]: https://github.com/vividus-framework/vividus-studio/issues/6 added feature request

**Pull request merged**